drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/AbstractKogitoDMNService.java

```diff
@@ -0,0 +1,469 @@
+        final List<JSITDRGElement> jsitdrgElements = jsitDefinitions.getDrgElement();
+        for (int i = 0; i < jsitdrgElements.size(); i++) {
+            final JSITDRGElement jsitdrgElement = Js.uncheckedCast(jsitdrgElements.get(i));
+            if (JSITInputData.instanceOf(jsitdrgElement)) {
+                JSITInputData jsitInputData = Js.uncheckedCast(jsitdrgElement);
+                final JSITInformationItem jsitInputDataVariable = jsitInputData.getVariable();
+                ClientDMNType type = getDMNTypeFromMaps(dmnTypesMap, JSITDMNElement.getOtherAttributesMap(jsitInputDataVariable));
+                checkTypeSupport(type, errorHolder, jsitInputData.getName());
+                visibleFacts.put(jsitInputData.getName(), createTopLevelFactModelTree(jsitInputData.getName(), type, hiddenFacts, FactModelTree.Type.INPUT));
+            } else if (JSITDecision.instanceOf(jsitdrgElement)) {
+                JSITDecision jsitDecision = Js.uncheckedCast(jsitdrgElement);
+                final JSITInformationItem jsitDecisionVariable = jsitDecision.getVariable();
+                ClientDMNType type = getDMNTypeFromMaps(dmnTypesMap, JSITDMNElement.getOtherAttributesMap(jsitDecisionVariable));
+                checkTypeSupport(type, errorHolder, jsitdrgElement.getName());
+                visibleFacts.put(jsitDecisionVariable.getName(), createTopLevelFactModelTree(jsitDecisionVariable.getName(), type, hiddenFacts, FactModelTree.Type.DECISION));
+            }
+        }
+        FactModelTuple toReturn = new FactModelTuple(visibleFacts, hiddenFacts);
+        errorHolder.getMultipleNestedCollection().forEach(toReturn::addMultipleNestedCollectionError);
+        errorHolder.getMultipleNestedObject().forEach(toReturn::addMultipleNestedObjectError);
+        return toReturn;
+    }
+
+    protected ClientDMNType getDMNTypeFromMaps(final Map<String, ClientDMNType> dmnTypesMap,
+                                               final Map<QName, String> source) {
+        String typeRef = source.get(TYPEREF_QNAME);
+        return dmnTypesMap.get(typeRef);
+    }
+
+    protected Map<String, ClientDMNType> getDMNTypesMap(final List<JSITItemDefinition> jsitItemDefinitions,
+                                                        final String nameSpace) {
+        Map<String, ClientDMNType> toReturn = new HashMap<>();
+        for (BuiltInType type : BuiltInType.values()) {
+            for (String name : type.getNames()) {
+                ClientDMNType feelPrimitiveType;
+                if (type == BuiltInType.CONTEXT) {
+                    feelPrimitiveType = new ClientDMNType(URI_FEEL, name, null, false, true, Collections.emptyMap(), type);
+                } else {
+                    feelPrimitiveType = new ClientDMNType(URI_FEEL, name, null, false, false, null, type);
+                }
+                toReturn.put(name, feelPrimitiveType);
+            }
+        }
+        jsitItemDefinitions.sort(getItemDefinitionComparator());
```

Comment:
Please add a comment to highlight the reason of this sort

Reply:
@danielezonca done